### PR TITLE
Add Github Actions for Windows

### PR DESCRIPTION
**Pull request by Luni-4**
`termion` can't be used on Windows

Thanks in advance for your review! :)

### CODE REVIEW

**Comment by lu-zero** [comment]
https://gitlab.redox-os.org/redox-os/termion/merge_requests/151 has some support that needs an overhaul.

**Comment by lu-zero** [comment]
We could consider https://github.com/mitsuhiko/console as an alternative?

**Comment by Luni-4** [comment]
> We could consider https://github.com/mitsuhiko/console as an alternative?

Yep, it could be a good alternative for now, waiting for the `termion`'s PR to land

**Comment by shssoichiro** [comment]
There are still some issues with the progress indicators anyway. I haven't found a good way to handle having multiple progress indicators that load one at a time. The main issue that still exists with the approach I'm using with termion is that once the segments get to the bottom of the terminal window, they start to overwrite each other and become unreadable.

There's an odd chance this might fix that issue as well? :crossed_fingers: And if it fixes Windows builds too, then I'm for it.

**Review by shssoichiro — APPROVED** [review]

**Comment by Luni-4** [comment]
> There's an odd chance this might fix that issue as well? crossed_fingers 

Hmm, I haven't verified that. Have you tested this kind of problem with a particular piece of code? If it's so, could you please share it? Thanks!

**Comment by shssoichiro** [comment]
It's reproducible in master but requires a long enough video that the progress indicators will vertically reach the bottom of the terminal. For example, I have an encode currently in progress:

Screenshot_20191203_091633

Once this hits around segment 60 (i.e. the bottom of the terminal), the remaining segments will start overwriting each other at the bottom of the terminal, instead of creating new lines.

I haven't had a chance to test it with this branch yet.

File: src/encode.rs [hunk 1]
```diff
@@ -77,8 +76,11 @@ pub fn perform_encode(keyframes: &[usize], opts: &CliOptions) -> Result<(), Box<
         }
     }
     thread_pool.join();
-    if is_tty(&stderr()) {
-        eprint!("{}", cursor::Goto(1, keyframes.len() as u16 + 5));
+    let term_err = Term::stderr();
+    if term_err.is_term() {
+        term_err
+            .move_cursor_to(1, (keyframes.len() as u16 + 5) as usize)
```
Comment (shssoichiro) [inline src/encode.rs]: It looks like in the console crate, the cursor indices are 0-indexed, while in termion they were 1-indexed. Would you mind correcting these?

**Review by shssoichiro — CHANGES_REQUESTED** [review]
This did not fix the issue with the progress bars hitting the bottom of the terminal window. I'll probably have to do something more creative to resolve that--i.e. not displaying 400 lines of progress. :thinking: 

Anyway, 1 small issue found, below.

**Comment by Luni-4** [comment]
> This did not fix the issue with the progress bars hitting the bottom of the terminal window. I'll probably have to do something more creative to resolve that--i.e. not displaying 400 lines of progress.

I haven't run it myself yet, I will do that later. But are you sure this problem isn't caused by your terminal? Have you tried to use another one?

**Comment by shssoichiro** [comment]
It's possible. I don't have a different terminal emulator to test with at the moment. Although, it's not caused by this patch (it's possibly just a limitation of this method of interacting with the terminal, which was the same in master).

**Review by shssoichiro — APPROVED** [review]

**Pull request merged**